### PR TITLE
Add on_exception to run() activity method if fail() method fails

### PR DESCRIPTION
**Pull request by Emersin**
We cannot add hard fail on timeout, because in case of timed out activity we still will receive successful response from execute_activity() method, but complete() method will fail with Exception "Unknown activity", to stop decider we try to run fail() method, but since activity timed out this method will also throw same Exception "Unknown activity". Raising Exception or returning "False" as result will not make decider to stop working. We decided to call on_exception() method if activity failed to call fail() method so we might find Exception in sentry since there was no logging when fail() method fails.

### CODE REVIEW

**Comment by papaonlegs** [comment]
@xethorn are you able to review?

File: garcon/activity.py [hunk 1]
```diff
@@ -320,8 +320,9 @@ def run(self, identity=None):
                     self.fail(reason=str(error)[:255])
                     if self.on_exception:
                         self.on_exception(self, error)
-                except:
-                    pass
+                except Exception as error2:  # noqa: E722
```
Comment (xethorn) [inline garcon/activity.py:323]: Is there a way to know the workflow has timed out before calling the `self.fail`?

**Comment by xethorn** [comment]
`master` branch is no longer used, can you rebase with `branch-1.0`? this will make the Github actions run.

File: garcon/activity.py (hunk 1 above)
Comment (Emersin) [reply garcon/activity.py:323]: Unfortunately we cannot, execute_activity() method will even return successful response like everything went OK, so it's basically impossible to know if activity timed out.

**Comment by Emersin** [comment]
> Two additional comments:
> 
> * `master` branch is no longer used, can you rebase with `branch-1.0`? this will make the Github actions run.
> * Please add some tests.

Tests added, target branch is branch-1.0

**Comment by xethorn** [comment]
LGTM, thanks!

**Pull request merged**